### DISCUSSION TITLE
Is it possible to keep the cache alive?

### DISCUSSION THREAD

**Discussion by luthianovolters** [espresso3389/pdfrx#224] [Q&A]
First, thanks for this awesome package, it's helping me a lot!

I am using this package on flutter web to display an uploaded PDF file with `Uint8List` on a new screen, but it always clears the cache after the first build of `PdfViewer.data()`.  

Note 1: it only clears the `Uint8List` in the class and in my custom controller with Provider, but the string name is kept in both;
Note 2: it only occurs with `PdfViewer`; `Image.memory()` does not clear the data;

Context of use: I'm looking for a solution to navigate to a new screen to display the PDF file and then return to the previous page so the user can save the file.

```
import 'dart:typed_data';
import 'package:flutter/material.dart';
import 'package:pdfrx/pdfrx.dart';

class ExampleViewer extends StatelessWidget {
  final String fileName;
  final Uint8List fileBytes;
  const ExampleViewer(
      {super.key, required this.fileName, required this.fileBytes});

  bool _isPDF() {
    return fileName.contains('pdf');
  }

  @override
  Widget build(BuildContext context) {
    return SafeArea(
        child: Scaffold(
            appBar: AppBar(title: const Text('Example Viewer')),
            body: _isPDF()
                ? PdfViewer.data(
                    fileBytes,
                    sourceName: fileName,
                  )
                : Center(child: Image.memory(fileBytes))));
  }
}
```

Need your help @espresso3389

**Comment by eduardogcorrea** [comment]
First, I'm learning (and I my language is portuguese, so, try not note my english faults).

I solve that using [flutter_cache_manager](https://pub.dev/packages/flutter_cache_manager). 

I load the PDF into cache_manager and get the PDF stored into the PDFRX. 
Bellow is my code (is under construction, this not have a proper loader / downloader indicator) but I think this maybe helpful.


```
import 'package:flutter_cache_manager/flutter_cache_manager.dart';

...

class _LeitorpdfState extends State<Leitorpdf> {
  final cacheManager = DefaultCacheManager();
  var pdfFileG = '';

  @override
  void initState() {
    super.initState();
    _loadPDF();
  }

...

 _loadPDF() async {
    var file = await cacheManager.getSingleFile(widget.url); //Get and load file from URL
    var fileKey = await cacheManager.getFileFromCache(widget.url); // Get filekey
    var fileFromCache = fileKey?.file.path; // Get file stored path

    setState(() {
      pdfFileG = fileFromCache!; // Set State from filepath
    });
  `}`
```

And then, use pdfFileG to show the file

```
PdfViewer.file(pdfFileG)
```
With this, the cache is stored in the [flutter_cache_manager](https://pub.dev/packages/flutter_cache_manager) and can be retrieved in any screen. Its just needed open an cache instance. 

As suggestion, the developers @espresso3389   maybe add a storage cache function / controllers.  This will be perfect.
